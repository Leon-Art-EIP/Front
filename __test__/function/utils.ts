import type { ReactTestRendererJSON } from "react-test-renderer";
// eslint-disable-next-line import/no-extraneous-dependencies
import renderer from "react-test-renderer";
import { expect } from "vitest";

function toJson(component: renderer.ReactTestRenderer): ReactTestRendererJSON {
  const result = component.toJSON();
  expect(result).toBeDefined();

  return result as ReactTestRendererJSON;
}

export function testSnapshot(element: JSX.Element): void {
  const component = renderer.create(element);
  const tree = toJson(component);
  expect(tree).toMatchSnapshot();
}
