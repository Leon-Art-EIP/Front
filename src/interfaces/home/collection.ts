export interface ICollection {
  artPublicationId: string;
  collectionName: string;
  isPublic: boolean;
}
