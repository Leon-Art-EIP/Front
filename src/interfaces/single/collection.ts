import { IArtPublication } from "../artPublication/artPublication";

export interface ICreateCollection {
  artPublicationId: string;
  collectionName: string;
  isPublic: boolean;
}

export interface ICollection {
  id: string;
  name: string;
  user: string; // id
  artPublications: string[]; // id
  isPublic: boolean;
}

export interface ICollectionArtsExtended extends Omit<ICollection, "artPublications"> {
  artPublications: IArtPublication[];
}
