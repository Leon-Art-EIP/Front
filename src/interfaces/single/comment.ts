export interface IComment {
  _id: string;
  userId: string;
  artPublicationId: string;
  text: string;
  createdAt: string;
}

export interface IDisplayComment {
  id: string;
  profilePicture: string;
  username: string;
  authorId: string;
  text: string;
  createdAt: string;
}

export interface IAddComment {
  msg: string;
  comment: Omit<IComment, "_id"> & {
    id: string;
  };
}
