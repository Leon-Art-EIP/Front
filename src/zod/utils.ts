export const getValueOrUndefined = (arg: unknown) =>
  arg === "" || arg === undefined || arg === null ? undefined : arg;
