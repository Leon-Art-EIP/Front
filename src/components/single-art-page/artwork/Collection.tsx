import { ICollectionArtsExtended } from "../../../interfaces/single/collection";
import Image from "next/image";
import { imageApi } from "../../../tools/variables";
import { cn } from "../../../tools/cn";

interface ICollectionProps {
  collection: ICollectionArtsExtended;
  handleSelectCollection(id: string): void;
  selected: boolean;
}

export default function Collection(props: ICollectionProps): JSX.Element {
  const artSources = props.collection.artPublications.map((art) => art.image);
  const firstPair: string[] = [];
  if (artSources.length > 0) firstPair.push(artSources[0]);
  if (artSources.length > 1) firstPair.push(artSources[1]);
  const secondPair: string[] = [];
  if (artSources.length > 2) secondPair.push(artSources[2]);
  if (artSources.length > 3) secondPair.push(artSources[3]);

  const pairs = [];
  if (firstPair.length > 0) pairs.push(firstPair);
  if (secondPair.length > 0) pairs.push(secondPair);

  const handleOnClick = () => {
    props.handleSelectCollection(props.collection._id);
  };

  return (
    <button
      className={cn(
        "flex flex-col items-center p-2 rounded-xl hover:cursor-pointer gap-2 border-4",
        props.selected ? "border-blue-300 bg-gray-300" : "hover:bg-secondaryGrey border-white"
      )}
      onClick={handleOnClick}
    >
      <div className="flex flex-col rounded-xl overflow-hidden w-48 h-48 bg-gray-300">
        {pairs.map((pair, pairIndex) => (
          <div key={`pair-${pairIndex}`} className="flex flex-wrap overflow-hidden flex-1">
            {pair.map((image, imageIndex) => (
              <Image
                key={`pair-${pairIndex}-image-${imageIndex}`}
                alt={`collection-picture-${pairIndex}-${imageIndex}`}
                src={`${imageApi}/${image}`}
                width={192}
                height={192}
                className="overflow-hidden flex-1"
              />
            ))}
          </div>
        ))}
        {artSources.length === 2 && <div className="flex-1 bg-gray-300" />}
      </div>
      <div className="text-2xl font-semibold">{props.collection.name}</div>
    </button>
  );
}
