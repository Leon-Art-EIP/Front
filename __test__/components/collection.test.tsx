import { test } from "vitest";
import { testSnapshot } from "../function/utils";
import Collection from "../../src/components/single-art-page/artwork/Collection";
import { ICollectionArtsExtended } from "../../src/interfaces/single/collection";

const collection: ICollectionArtsExtended = {
  _id: "1",
  name: "Collection name",
  artPublications: [
    {
      _id: "1",
      userId: "1",
      image: "image",
      artType: "artType",
      name: "name",
      description: "description",
      isForSale: "true",
      isSold: false,
      price: 100,
      location: "location",
      likes: ["1"],
      totalLikes: 1,
      comments: ["1"],
    },
  ],
  isPublic: true,
  userId: "1",
};

test("Collection snapshot", () => {
  testSnapshot(<Collection handleSelectCollection={() => {}} selected={false} collection={collection} />);
});

test("Collection selected snapshot", () => {
  testSnapshot(<Collection handleSelectCollection={() => {}} selected={true} collection={collection} />);
});
