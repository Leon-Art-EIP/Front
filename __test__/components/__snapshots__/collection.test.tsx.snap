// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Collection selected snapshot 1`] = `
<button
  className="flex flex-col items-center p-2 rounded-xl hover:cursor-pointer gap-2 border-4 border-blue-300 bg-gray-300"
  onClick={[Function]}
>
  <div
    className="flex flex-col rounded-xl overflow-hidden w-48 h-48 bg-gray-300"
  >
    <div
      className="flex flex-wrap overflow-hidden flex-1"
    >
      <img
        alt="collection-picture-0-0"
        className="overflow-hidden flex-1"
        height={192}
        src="undefined/api/image"
        width={192}
      />
    </div>
  </div>
  <div
    className="text-2xl font-semibold"
  >
    Collection name
  </div>
</button>
`;

exports[`Collection snapshot 1`] = `
<button
  className="flex flex-col items-center p-2 rounded-xl hover:cursor-pointer gap-2 border-4 hover:bg-secondaryGrey border-white"
  onClick={[Function]}
>
  <div
    className="flex flex-col rounded-xl overflow-hidden w-48 h-48 bg-gray-300"
  >
    <div
      className="flex flex-wrap overflow-hidden flex-1"
    >
      <img
        alt="collection-picture-0-0"
        className="overflow-hidden flex-1"
        height={192}
        src="undefined/api/image"
        width={192}
      />
    </div>
  </div>
  <div
    className="text-2xl font-semibold"
  >
    Collection name
  </div>
</button>
`;
